src/main/java/mServer/crawler/AddToFilmlist.java

```diff
@@ -5,176 +5,270 @@
         Log.sysLog("");
-        size = listeEinsortieren.size();
+    }
+
+    private void performUrlSearch(HashSet<Hash> hash, final int size) {
+        vonListe.parallelStream().forEach(f -> {
+            synchronized (hash) {
+                hash.add(f.getHashValueUrl());
+            }
+        });
 
-        // ==============================================
-        // nach "URL" suchen
-        EtmPoint performancePointUrlSuchen = etmMonitor.createPoint("AddToFilmlist:addOldList#UrlSuchen");
-        vonListe.stream().forEach((f) -> hash.add(DatenFilm.getUrl(f)));
-        listeEinsortieren.removeIf((f) -> hash.contains(DatenFilm.getUrl(f)));
+        listeEinsortieren.removeIf((f) -> hash.contains(f.getHashValueUrl()));
         hash.clear();
-        performancePointUrlSuchen.collect();
 
         Log.sysLog("===== Liste einsortieren URL =====");
         Log.sysLog("Liste einsortieren, Anzahl: " + size);
         Log.sysLog("Liste einsortieren, entfernt: " + (size - listeEinsortieren.size()));
         Log.sysLog("Liste einsortieren, noch einsortieren: " + listeEinsortieren.size());
         Log.sysLog("");
+    }
 
-        Duration.staticPing("AddOld-2");
+    /**
+     * Remove links which don´t start with http.
+-    * Remove old film entries which are smaller than MIN_SIZE_ADD_OLD.
+     */
+    private void performInitialCleanup() {
+        listeEinsortieren.removeIf(f -> !f.arr[DatenFilm.FILM_URL].toLowerCase().startsWith("http"));
+        listeEinsortieren.removeIf(f -> {
+            String groesse = f.arr[DatenFilm.FILM_GROESSE];
+            if (groesse.isEmpty())
+                return false;
+            else {
+                return Long.parseLong(groesse) < MIN_SIZE_ADD_OLD;
+            }
 
-        int count = 0;
-        final int MAX_THREAD = 30;
-        final int MAX_WAIT_TIME = 450; // 450*2=900s -> 15 Minuten
-        //final int MAX_WAIT_TIME = 10; // 10*2=20s
-        stopOld = false;
-        size = listeEinsortieren.size();
+        });
+    }
+
+    private void startThreads() {
+        final OkHttpClient client = MVHttpClient.getInstance().getReducedTimeOutClient();
+        for (int i = 0; i < NUMBER_OF_THREADS; ++i) {
+            ImportOldFilmlistThread t = new ImportOldFilmlistThread(listeEinsortieren, client);
+            t.setName("ImportOldFilmlistThread Thread-" + i);
+            threadList.add(t);
+            t.start();
+        }
+    }
 
-        // Rest nehmen wir wenn noch online
-        for (int i = 0; i < MAX_THREAD; ++i) {
-            new Thread(new AddOld(listeEinsortieren)).start();
+    private void stopThreads() {
+        if (Config.getStop()) {
+            for (ImportOldFilmlistThread t : threadList)
+                t.interrupt();
+            for (ImportOldFilmlistThread t : threadList) {
+                try {
+                    t.join();
+                } catch (InterruptedException ignored) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either re-interrupt this method or rethrow the "InterruptedException". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2142)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either remove or fill this block of code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00108)